lib/saluki-components/src/transforms/dogstatsd_prefix_filter/mod.rs

```diff
@@ -0,0 +1,349 @@
+            let new_metric_name = if self.is_excluded(metric_name) {
+                metric.context().name().clone()
+            } else {
+                MetaString::from(format!("{}{}", self.metric_prefix, metric_name))
+            };
+
+            if self.blocklist.contains(&new_metric_name) {
+                debug!("Metric {} excluded due to blocklist.", new_metric_name);
+                return None;
+            }
+
+            // Update metric with new name.
+            let new_context = metric.context().with_name(new_metric_name);
+            let existing_context = metric.context_mut();
+            *existing_context = new_context;
+        }
+        Some(metric)
+    }
+
+    fn is_excluded(&self, metric_name: &str) -> bool {
+        !self.metric_prefix.is_empty()
+            && self
+                .metric_prefix_blacklist
+                .iter()
+                .any(|prefix| metric_name.starts_with(prefix))
+    }
+}
+
+#[async_trait]
+impl Transform for DogstatsDPrefixFilter {
+    async fn run(mut self: Box<Self>, mut context: TransformContext) -> Result<(), GenericError> {
+        let mut health = context.take_health_handle();
+        health.mark_ready();
+
+        debug!("Agent metric name filter transform started.");
+
+        loop {
+            select! {
+                _ = health.live() => continue,
+                maybe_events = context.event_stream().next() => match maybe_events {
+                    Some(events) => {
+                        let mut buffered_forwarder = context.forwarder().buffered().expect("default output must always exist");
+
+                        for event in events {
+                            if let Some(metric) = event.try_into_metric() {
+                                if let Some(new_metric) = self.enrich_metric(metric) {
+                                    if let Err(e) = buffered_forwarder.push(saluki_event::Event::Metric(new_metric)).await {
+                                        error!(error = %e, "Failed to forward event.");
+                                    }
+                                }
+                            }
+                        }
+
+                        if let Err(e) = buffered_forwarder.flush().await {
+                            error!(error = %e, "Failed to forward events.");
+                        }
+
+                    },
+                    None => break,
+                },
+            }
+        }
+
+        debug!("Agent metric name transform stopped.");
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use saluki_context::Context;
+
+    use super::*;
+
+    #[test]
+    fn test_metric_prefix_add() {
+        let filter = DogstatsDPrefixFilter {
+            metric_prefix: "foo".to_string(),
+            metric_prefix_blacklist: vec![],
+            blocklist: Blocklist::default(),
+        };
+        let context = Context::from_static_parts("bar", &[]);
+        let metric = Metric::gauge(context, 1.0);
```

Comment:
FWIW, you can just do `Metric::gauge("bar", 1.0);` since there's an implementation to create a `Context` purely from a single static string, which implies no tags.